### PR TITLE
feat: preview spec changes for different helper libs

### PR DESCRIPTION
**Pull request by shrutiburman**
# Fixes [DII-422](https://issues.corp.twilio.com/browse/DII-422), [DII-423](https://issues.corp.twilio.com/browse/DII-423)

This PR: 

- Adds support for Go and Terraform to skip the preview subdomain while generating code. 
- Allows Java and other helper libraries to support `preview` spec.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [x] Verify affected language:
    - [x] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: scripts/build_twilio_library.py [hunk 1]
```diff
@@ -28,26 +28,31 @@ def generate(openapi_spec_path: str, output_path: str, language: str, domain: st
...
+                      f'org.openapitools.codegen.OpenAPIGenerator generate -g {to_generate} -i {full_path} ' \
+                      f'-o {output_path} ' \
+                      f'> /dev/null'  # Suppress stdout
+    print(f'Generating {output_path} from {full_path}')
+    os.system(command)
+
 def get_domain_info(oai_spec_location: str, domain: str, is_file: bool = False) -> Tuple[str, str, str]:
     full_path = oai_spec_location if is_file else os.path.join(oai_spec_location, domain)
-    parts = re.split(r'twilio_(.+?)_(v\d+)', domain, flags=re.IGNORECASE)
+    parts = re.split(r'twilio_(.+?)_?(v\d+)?\.', domain, flags=re.IGNORECASE)
```
Comment (shrutiburman) [inline scripts/build_twilio_library.py:53]: [self] :`preview` spec filename doesn't have version which needed to be handled in the regex.

File: scripts/build_twilio_library.py [hunk 2]
```diff
@@ -28,26 +28,31 @@ def generate(openapi_spec_path: str, output_path: str, language: str, domain: st
...
     domain_name = parts[1]
-    api_version = parts[2]
+    api_version = parts[2] if parts[2] != None else ''
```
[the remark this answers is not in the document]
Comment (childish-sambino) [reply scripts/build_twilio_library.py:55]: nit: could just be `api_version = parts[2] or ''`

File: scripts/build_twilio_library.py [hunk 3]
```diff
@@ -28,26 +28,31 @@ def generate(openapi_spec_path: str, output_path: str, language: str, domain: st
...
     output_path = os.path.join(output_path, sub_dir, domain_name)
     if language in {'go', 'terraform'}:
         output_path = os.path.join(output_path, api_version)
-    command = f'cd {parent_dir} && java -cp ./openapi-generator-cli.jar:target/twilio-openapi-generator.jar ' \
-              f'org.openapitools.codegen.OpenAPIGenerator generate -g {to_generate} -i {full_path} ' \
-              f'-o {output_path} ' \
-              f'> /dev/null'  # Suppress stdout
-    print(f'Generating {output_path} from {full_path}')
-    os.system(command)
+    if(is_domain_irrelevant == False):
```
Comment (childish-sambino) [inline scripts/build_twilio_library.py:36]: nit: should not wrap condition in parenthesis.

File: scripts/build_twilio_library.py [hunk 4]
```diff
@@ -28,26 +28,31 @@ def generate(openapi_spec_path: str, output_path: str, language: str, domain: st
     parent_dir = Path(__file__).parent.parent
 
     to_generate = 'terraform-provider-twilio' if language == 'terraform' else f'twilio-{language}'
+    is_domain_irrelevant = True if language in {'go', 'terraform'} and domain_name == 'preview' else False
```
Comment (childish-sambino) [inline scripts/build_twilio_library.py:31]: nit: `True if` and `else False` not needed as it's implicit.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**